src/coreComponents/constitutive/solid/DelftEgg.hpp

```diff
@@ -0,0 +1,713 @@
+                                  real64 ( & stress )[6],
+                                  real64 ( & stiffness )[6][6] ) override final;
+  
+  
+  GEOSX_HOST_DEVICE
+  virtual void SaveConvergedState( localIndex const k,
+                                   localIndex const q ) override final;
+    
+private:
+
+  /// A reference to the ArrayView holding the shear modulus for each element.
+  arrayView1d< real64 const > const m_shearModulus;
+  
+  /// A reference to the ArrayView holding the reference pressure (p0) for each element.
+  arrayView2d< real64 const > const m_refPressure;
+
+    /// A reference to the ArrayView holding the shear modulus for each element.
+  arrayView2d< real64 const > const m_refStrainVol;
+
+  /// A reference to the ArrayView holding the recompression index for each element.
+  arrayView1d< real64 const > const m_recompressionIndex;
+  
+  /// A reference to the ArrayView holding the virgin compression index for each element.
+  arrayView1d< real64 const > const m_virginCompressionIndex;
+  
+  /// A reference to the ArrayView holding the slope of the critical state line for each element.
+  arrayView1d< real64 const > const m_cslSlope;
+  
+  /// A reference to the ArrayView holding the shape parameter for each integration point
+  arrayView1d< real64 const > const m_shapeParameter;
+
+  /// A reference to the ArrayView holding the old preconsolidation pressure for each integration point
+  arrayView2d< real64 > const m_oldPreConsolidationPressure;
+
+  /// A reference to the ArrayView holding the new preconsolidation pressure for each integration point
+  arrayView2d< real64 > const m_newPreConsolidationPressure;
+  
+  /// A reference to the ArrayView holding the new stress for each integration point
+  arrayView3d< real64, solid::STRESS_USD > const m_newStress;
+  
+  /// A reference to the ArrayView holding the old stress for each integration point
+  arrayView3d< real64, solid::STRESS_USD > const m_oldStress;
+};
+
+///////////////////////// new proposal /////////////////////////////////////////
+
+GEOSX_HOST_DEVICE
+GEOSX_FORCE_INLINE
+void DelftEggUpdates::SmallStrainUpdate( localIndex const k,
+                                         localIndex const q,
+                                         real64 const ( & strainIncrement )[6],
+                                         real64 ( & stress )[6],
+                                         real64 ( & stiffness )[6][6] )
+{
+  real64 const oldPc  = m_oldPreConsolidationPressure[k][q]; //pre-consolidation pressure
+  real64 const mu     = m_shearModulus[k];
+  real64 const p0     = m_refPressure[k][q];
+
+  real64 const eps_v0 = m_refStrainVol[k][q];
+  real64 const M      = m_cslSlope[k];
+  real64 const Cr     = m_recompressionIndex[k];
+  real64 const Cc     = m_virginCompressionIndex[k];
+  real64 const alpha  = m_shapeParameter[k];
+
+  real64        pc    = oldPc; 
+  real64 bulkModulus  = -p0/Cr;
```

Comment:
Poisson ratio is missing? I checked the original paper of Delft Egg model, it is written that: K = -nu*p'/kappa. If Cr is kappa then Poisson ratio nu is missing.